frontend/src/metabase/dashboard/reducers-typed.ts

```diff
@@ -413,3 +408,48 @@ export const loadingDashCards = createReducer(
       }));
   },
 );
+
+export const dashcardData = createReducer(
+  INITIAL_DASHBOARD_STATE.dashcardData,
+  builder => {
+    builder
+      .addCase(initialize, (state, action) => {
+        const { clearCache = true } = action.payload ?? {};
+        return clearCache ? {} : state;
+      })
+      .addCase(fetchCardDataAction.fulfilled, (state, action) => {
+        const { dashcard_id, card_id, result } = action.payload ?? {};
+        if (dashcard_id && card_id) {
+          return assocIn(state, [dashcard_id, card_id], result);
+        }
+      })
+      .addCase(clearCardData, (state, action) => {
+        const { cardId, dashcardId } = action.payload;
+        return dissocIn(state, [dashcardId, cardId]);
```

Comment:
ah! no this isn't a mistake. We were using `assocIn` but assigning the value to `undefined`, which typescript hated. So I changed things to `dissocIn` since I think that is basically just `assocIn(state, [dashcardId, cardId], undefined)`

Reply:
That makes sense. Man, the `assocIn` without the last param sucks. It's really hard to understand at a glance.

Reply:
I'm so glad that trying to use `assocIn` without that last parameter makes Typescript angry. It'll really help when we get all of this stuff over